x/distribution/abci_test.go

```diff
@@ -0,0 +1,109 @@
+package distribution_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/teststaking"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/assert"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+const (
+	totalValidators  = 6
+	lazyValidatorIdx = 2
+	power            = 100 / totalValidators
+)
+
+var (
+	valTokens                = sdk.TokensFromConsensusPower(50, sdk.DefaultPowerReduction)
+	validatorCommissionRates = stakingtypes.NewCommissionRates(sdk.OneDec(), sdk.OneDec(), sdk.OneDec())
+)
+
+type validator struct {
+	addr   sdk.ValAddress
+	pubkey cryptotypes.PubKey
+	votes  []abci.VoteInfo
+}
+
+// Context in https://github.com/cosmos/cosmos-sdk/issues/9161
+func Test_VerifyProposerRewardAssignement(t *testing.T) {
+	a := assert.New(t)
```

Comment:
See my comment in the other similar PR :)

Reply:
Done!